lib/rest/RequestHandler.js

```diff
@@ -60,6 +61,10 @@ class RequestHandler {
         const _stackHolder = {}; // Preserve async stack
         Error.captureStackTrace(_stackHolder);
 
+        if(body && this.invalidHeaderRegex.test(body.reason)) {
```

Comment:
you might want to check if body.reason exists

Reply:
`undefined` always returns false, so not needed